src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java

```diff
@@ -152,6 +217,379 @@ void shouldFetchEmptyUserTypes() {
         Assertions.assertArrayEquals(actualResponse.getUserTypes().toArray(), userTypes.toArray());
     }
 
+
+    @ParameterizedTest
+    @ValueSource(strings = {"BFA1,BFA2","BFA1"})
+    void should_return_staffProfiles_search_with_service_code_status_code_200(String serviceId) {
+
+        List<ServiceResponse> services = List.of(ServiceResponse.builder().serviceCode("BFA1").service("avd").build());
+        searchResponse.setServices(services);
+
+        searchReq = SearchRequest.builder()
+                .serviceCode(serviceId)
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(List.of(searchResponse), null, HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response =
+                staffRefDataController.searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"s*_"})
+    void should_return_staffProfiles_search_with_invalid_service_code_400(String serviceId) {
+        searchReq = SearchRequest.builder()
+                .serviceCode(serviceId)
+                .build();
+        assertThrows(InvalidRequestException.class, () -> staffRefDataController
```

Comment:
can you check for expected error message

Reply:
changes done